apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/monetary/service/AssetServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.monetary.service;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.util.stream.Stream;
+
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.converter.IteratorToStreamConverter;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.DbKey;
+import com.apollocurrency.aplwallet.apl.core.db.service.BlockChainInfoService;
+import com.apollocurrency.aplwallet.apl.core.monetary.dao.AssetTable;
+import com.apollocurrency.aplwallet.apl.core.monetary.model.Asset;
+import com.apollocurrency.aplwallet.apl.core.transaction.messages.ColoredCoinsAssetIssuance;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class AssetServiceImpl implements AssetService {
+
+    private AssetTable assetTable;
+    private BlockChainInfoService blockChainInfoService;
+    private AssetDeleteService assetDeleteService;
+    private IteratorToStreamConverter<Asset> assetIteratorToStreamConverter =
+        new IteratorToStreamConverter<>();
+
+    @Inject
+    public AssetServiceImpl(AssetTable assetTable,
+                            BlockChainInfoService blockChainInfoService,
+                            AssetDeleteService assetDeleteService,
+                            IteratorToStreamConverter<Asset> assetIteratorToStreamConverter // for unit tests mostly
+    ) {
+        this.assetTable = assetTable;
+        this.blockChainInfoService = blockChainInfoService;
+        this.assetDeleteService = assetDeleteService;
+        if (assetIteratorToStreamConverter != null) { // for unit tests
```

Comment:
I'd prefer to not add additional conditions only for unit test usage, is it may be avoided?

Reply:
That simplifies writing mock tests. Another replacement solutions can be more complicated then this one.